pkg/resourcecollector/virtualmachine.go

```diff
@@ -1,10 +1,101 @@
 package resourcecollector
 
 import (
+	"strings"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// Transform dataVolume: to persistentVolumeClaim:
+//	       name:	      claimName:
+func transformDataVolume(input map[string]interface{}, name string ) map[string]interface{} {
+
+	output := make(map[string]interface{})
+
+	for key, val := range input {
+		if key == name {
+			output["claimName"] = val
+			continue
+		}
+		output[key] = val
+	}
+	return output
+}
+
+// Utility function for virtualMachine template parsing logic
+func parseMap(input map[string]interface{}, output map[string]interface{}, is_key_present *bool) {
+
+	for mKey, mVal := range input {
+		switch mKey {
+		case "macAddress":
+			*is_key_present = true
+			continue
+		case "dataVolume":
+			*is_key_present = true
+			switch typeval := mVal.(type) {
+			case map[string]interface{}:
+				output["persistentVolumeClaim"] =  transformDataVolume(typeval, "name")
+				continue
+			}
+		}
+		output[mKey] = mVal
+	}
+}
+
+// Path parsing function for virtualMachine template.
+func transformPath(object runtime.Unstructured, path []string) error {
+
+        content := object.UnstructuredContent()
+        data, _, err := unstructured.NestedSlice(content, path...)
+        if err != nil {
+		return err
+        }
+        found := false
+        interf := make([]interface{}, 0, 1)
+        for _, val := range data {
+		switch typeval := val.(type) {
+		case map[string]interface{}:
+			newMap := make(map[string]interface{})
```

Comment:
As discussed, can we try `RemoveNestedField()` and passing macAddress field

Reply:
Tried and this doesnt work. Since macAddress is not a nestedField.